Source/DiabloUI/dialogs.cpp

```diff
@@ -166,56 +165,50 @@ void Init(const char *text, const char *caption, bool error, bool renderBehind)
 		vecOkDialog.push_back(std::make_unique<UiImage>(&dialogArt, rect1));
 
 		SDL_Rect rect2 = { (Sint16)(PANEL_LEFT + 200), (Sint16)(UI_OFFSET_Y + 211), 240, 80 };
-		vecOkDialog.push_back(std::make_unique<UiText>(text, rect2, UiFlags::AlignCenter));
+		vecOkDialog.push_back(std::make_unique<UiText>(text, rect2, UiFlags::AlignCenter | UiFlags::ColorDialogWhite));
 
 		SDL_Rect rect3 = { (Sint16)(PANEL_LEFT + 265), (Sint16)(UI_OFFSET_Y + 265), SML_BUTTON_WIDTH, SML_BUTTON_HEIGHT };
 		vecOkDialog.push_back(std::make_unique<UiButton>(&SmlButton, _("OK"), &DialogActionOK, rect3));
 	} else {
 		SDL_Rect rect1 = { (Sint16)(PANEL_LEFT + 127), (Sint16)(UI_OFFSET_Y + 100), 385, 280 };
 		vecOkDialog.push_back(std::make_unique<UiImage>(&dialogArt, rect1));
 
-		SDL_Color color = { 255, 255, 0, 0 };
 		SDL_Rect rect2 = { (Sint16)(PANEL_LEFT + 147), (Sint16)(UI_OFFSET_Y + 110), 345, 20 };
-		vecOkDialog.push_back(std::make_unique<UiText>(text, rect2, UiFlags::AlignCenter, color));
+		vecOkDialog.push_back(std::make_unique<UiText>(text, rect2, UiFlags::AlignCenter | UiFlags::ColorDialogYellow));
 
 		SDL_Rect rect3 = { (Sint16)(PANEL_LEFT + 147), (Sint16)(UI_OFFSET_Y + 141), 345, 190 };
-		vecOkDialog.push_back(std::make_unique<UiText>(caption, rect3, UiFlags::AlignCenter));
+		vecOkDialog.push_back(std::make_unique<UiText>(caption, rect3, UiFlags::AlignCenter | UiFlags::ColorDialogWhite));
 
 		SDL_Rect rect4 = { (Sint16)(PANEL_LEFT + 264), (Sint16)(UI_OFFSET_Y + 335), SML_BUTTON_WIDTH, SML_BUTTON_HEIGHT };
 		vecOkDialog.push_back(std::make_unique<UiButton>(&SmlButton, _("OK"), &DialogActionOK, rect4));
 	}
 
 	if (!renderBehind) {
+		ArtBackground.Unload();
 		LoadBackgroundArt("ui_art\\black.pcx");
 		if (ArtBackground.surface == nullptr) {
 			LoadFallbackPalette();
+			if (SDL_ShowCursor(SDL_ENABLE) <= -1)
+				Log("{}", SDL_GetError());
 		}
 	}
-	SetFadeLevel(256);

```

Comment:
Why are there changes to the fade level and background art in this file?

Reply:
I fixes a bunch of edge cases for the error dialog, they where more evident when working on the font code as they mostly manifest when there is a failure early on.

Specifically the fade level should always start out black as each dialog screen will slowly fade to full bright. The line here was causing the screen to flicker when the error log appeared.